src/engine/platform/amiga.cpp

```diff
@@ -95,13 +95,13 @@ void DivPlatformAmiga::acquire(short* bufL, short* bufR, size_t start, size_t le
             }
           } else {
             DivSample* s=parent->getSample(chan[i].sample);
-            if (s->samples>0) {
-              if (chan[i].audPos<s->samples) {
+            if (s->getEndPosition()>0) {
```

Comment:
I suggest you revert this (on all systems). We need this comparison to be as fast as possible and calling a function to do it is definitely not fast.